PinKit/Sources/PinKit/My Data/Cells/MusicCellView.swift

```diff
@@ -36,40 +39,53 @@ struct MusicCellView: View {
                 }
             }
         } else {
-            LabeledContent {
-                if let id = event.albumArtUuid {
-                    WebImage(url: makeAlbumURL(id: id)) { image in
-                        image
-                            .resizable()
-                            .scaledToFit()
-                            .frame(width: 60, height: 60)
-                            .clipShape(RoundedRectangle(cornerRadius: 8))
-                    } placeholder: {
-                        RoundedRectangle(cornerRadius: 8)
-                            .fill(.bar)
-                            .frame(width: 60, height: 60)
-                            .overlay(ProgressView())
+            Button {
+                if let trackId = event.trackID {
+                    if let trackUrl = makeTidalTrackURL(trackID: trackId) {
+                        openURL(trackUrl)
```

Comment:
right above this (line 38) you should do: `else if let trackId = event.trackID`, otherwise we have a button that doesn't do anything.